### ISSUE TITLE
Rule request: enforce the use of template strings over standard concatenation

### ISSUE THREAD

**Issue by SamuelBolduc** [eslint/eslint#3014]
It was suggested in #1617 to add a rule to enforce the use of string templates. As far as I know, this has not been implemented yet. 

I was wondering if someone thought about this or began work on this rule?

**Comment by nzakas** [comment]
You can use the quotes rule stuff the backticks setting: http://eslint.org/docs/rules/quotes

**Comment by SamuelBolduc** [comment]
I don't think it was related - what I was talking about was to enforce ES6 string templates instead of standard concatenation. 

I'm looking for a rule that would give me an error for this : 

`var myString = 'Hello, my name is ' + name + '!';`

But would accept this syntax :

`var myString = 'Hello, my name is ${name}!';`

I don't think this is covered by the quotes rule.

**Comment by ilyavolodin** [comment]
I agree, quotes rule will not cover this request.

**Comment by IanVS** [comment]
So you would want a rule which could enforce:

```js
// No error
var myString = "Hello, my name is Bob"; 
var myString = 'Hello, my name is Bob'; 

// Error
var myString = "Hello, my name is " + name; 
```

**Comment by ilyavolodin** [comment]
Basically this rule should flag string concatenation and suggest to use string templates instead.

**Comment by SamuelBolduc** [comment]
Exactly!

```
// No error
var myString = "Hello, my name is Bob"; 
var myString = 'Hello, my name is Bob'; 
var myString = 'Hello, my name is ${name}';

// Error
var myString = "Hello, my name is " + name;
 ```

This way, when using it along with other ES6-enforcing rules, you can learn to use the advantages of ES6 much quicker!

**Comment by nzakas** [comment]
That's fine by me.

@SamuelBolduc in the future, please be sure to include sample code in your description along with what sort of warning you'd expect. We get a lot of feature requests and it's easy to overlook a word or two. Here's what we'd like for new rules: http://eslint.org/docs/developer-guide/contributing.html#new-rules

**Comment by mysticatea** [comment]
I will work on this.

**Comment by xjamundx** [comment]
Awesome!
